python_modules/libraries/dagster-components/dagster_components_tests/utils.py

```diff
@@ -108,23 +109,32 @@ def inject_component(
 @contextlib.contextmanager
 def create_project_from_components(
     *src_paths: str, local_component_defn_to_inject: Optional[Path] = None
-) -> Iterator[Path]:
+) -> Iterator[tuple[Path, str]]:
     """Scaffolds a project with the given components in a temporary directory,
     injecting the provided local component defn into each component's __init__.py.
     """
+    location_name = f"my_location_{str(random.random()).replace('.', '')}"
```

Comment:
another bit of test-code shenanigans, `importlib.invalidate_caches()` did not seem to fully invalidate the cache so when you ran multiple tests at once that all pointed at a location named "my_location", you would get incorrect outputs.